.travis.yml

```diff
@@ -1,35 +1,34 @@
-sudo: false
-os:
-  - linux
-
-language: node_js
-node_js:
-  - "8"
-
-addons:
-  apt:
-    sources:
-    - ubuntu-toolchain-r-test
-    packages:
-    - g++-4.8
+dist: xenial
 
 branches:
   # We need to whitelist the branches which we want to have "push" automation.
   # Pull request automation is not constrained to this set of branches.
   only:
   - master
 
-env:
-  global:
-    - DISPLAY=:99.0
-  matrix:
-    - CXX=g++-4.8 TEST_SUITE=test
-
 matrix:
   fast_finish: true
   include:
-    - os: linux
+    - language: node_js
       node_js: "8"
-      env: CXX=g++-4.8 TEST_SUITE=test
+      env: TEST_SUITE=test
+      script:
+        - npm run $TEST_SUITE
 
-script: npm run $TEST_SUITE
\ No newline at end of file
+    - language: rust
+      rust: "1.27.2"
+      env: EMSDK_VERSION=sdk-1.38.21-64bit
+      install:
+        #- rustup component add rustfmt
+        # install emsdk
+        - git clone https://github.com/emscripten-core/emsdk.git
+        - ./emsdk/emsdk install $EMSDK_VERSION
+        - ./emsdk/emsdk activate $EMSDK_VERSION
+        - rustup target add asmjs-unknown-emscripten
+      script:
+        # the useful way to run rustfmt
+        - cargo fmt --all -- --check
+        # activate Emscripten
+        - source ./emsdk/emsdk_env.sh
+        - cargo build
```

Comment:
If you look at the makefile, `cargo build` requires the asmjs target to be specified. You are better off just running `make` and no `cargo build`.

Reply:
@axic why https://github.com/ethereumjs/rustbn.js/issues/25 then says to run `cargo build` and Makefle as two separate steps?